app/controllers/concerns/foreman/controller/registration.rb

```diff
@@ -29,8 +28,8 @@ def global_registration_vars
       operatingsystem: operatingsystem,
       url_host: registration_url.host,
       registration_url: registration_url,
-      setup_insights: global_setup_insights(host_config_params),
-      setup_remote_execution: global_setup_remote_execution(host_config_params),
+      setup_insights: ActiveRecord::Type::Boolean.new.deserialize(params['setup_insights']),
```

Comment:
Where comes the default (Hostgroup or OS parameter value) in play now? :thinking:

Reply:
In the host registration template, specifically in the snippets [remote_execution_ssh_keys](https://github.com/theforeman/foreman/blob/develop/app/views/unattended/provisioning_templates/snippet/remote_execution_ssh_keys.erb) & [insights](https://github.com/theforeman/foreman/blob/develop/app/views/unattended/provisioning_templates/snippet/insights.erb). We check if the params are true and if yes than we render the snippets.

Reply:
maybe I'm missing something (quite possibly), but I can't find use of `setup_remote_execution` in any of the templates, but global_registration and that one is not the one, right?

Reply:
`setup_rex|insights` are called in [api/v2/registration_controller](https://github.com/theforeman/foreman/blob/develop/app/controllers/api/v2/registration_controller.rb#L76), when we generate **host** registration template.

The WF is like this:
* `GET /register?setup_rex=true`
* -> Render Global registration template, with `curl -X POST /register` to render host rt (and passing `setup_rex=true`), see [template](https://github.com/theforeman/foreman/blob/develop/app/views/unattended/provisioning_templates/registration/global_registration.erb#L49)
* Run commands from Global RT
* Call for Host registration template
* -> Setup host & its parameters
* -> Render template
* `template-content | bash`